File: epi_judge_java/epi/SnakeString.java
package epi;

import epi.test_framework.EpiTest;
import epi.test_framework.GenericTest;

public class SnakeString {

    @EpiTest(testDataFile = "snake_string.tsv")
    public static String snakeString(String s) {
        StringBuilder[] result = new StringBuilder[3];
        for (int i = 0; i < 3; i++) {
            result[i] = new StringBuilder();
        }
        for (int i = 0; i < s.length(); i++) {
            if (i % 4 == 1) {
                result[0].append(s.charAt(i));
            } else if (i % 4 == 3) {
                result[2].append(s.charAt(i));
            } else {
                result[1].append(s.charAt(i));
            }
        }

        return result[0].append(result[1]).append(result[2]).toString();
    }

    public static void main(String[] args) {
        System.exit(
                GenericTest
                        .runFromAnnotations(args, "SnakeString.java",
                                new Object() {
                                }.getClass().getEnclosingClass())
                        .ordinal());
    }
}

File: epi_judge_cpp/snake_string.cc
#include <string>

#include "test_framework/generic_test.h"

using std::string;

string SnakeString(const string &s) {
    string s1;
    string s2;
    string s3;

    for (int i = 0; i < s.size(); i++) {
        if (i % 4 == 1) {
            s1 += s[i];
        } else if (i % 4 == 3) {
            s3 += s[i];
        } else {
            s2 += s[i];
        }
    }
    return s1 + s2 + s3;
}

int main(int argc, char *argv[]) {
    std::vector<std::string> args{argv + 1, argv + argc};
    std::vector<std::string> param_names{"s"};
    return GenericTestMain(args, "snake_string.cc", "snake_string.tsv",
                           &SnakeString, DefaultComparator{}, param_names);
}
